### ISSUE TITLE
can't start with custom chain config

### ISSUE THREAD

**Issue by splix** [ethereumproject/go-ethereum#330]
```
$ geth --datadir /data --rpc --rpcaddr 0.0.0.0 --chain ./ecip1017chain.json
F0810 00:45:30.335703 cmd/geth/flag.go:663] chain config not found: /data/ecip1017chain.json/chain.json
		It looks like you haven't set up your custom chain yet...
		Here's a possible workflow for that:

		$ geth --chain morden dump-chain-config /data/ecip1017chain.json/chain.json
		$ sed -i.bak s/morden/ecip1017chain.json/ /data/ecip1017chain.json/chain.json
		$ vi /data/ecip1017chain.json/chain.json # <- make your customizations
```

**Comment by whilei** [comment]
I'm guessing the chain config file is invalid, so it's reverting to trying argument as literal chain name. You can debug with `--verbosity 6`. 

I agree that this current error output is confusing. In order to give better errors, geth needs to be able to decide whether or not the user is expecting a chain identity-name OR file configuration. To decide this, either geth should differentiate expected chain configuration types by
-  if there's a `/` in the argument
  + `--chain=./ecip1017.json` --> file config
  + `--chain=./ecip10117config` --> file config
  + `--chain=ecip1017` --> identity config
  
- if the argument's final extension is `.json`. I'm a little hesitant to enforce file name conventions...

**Comment by whilei** [comment]
Please see #333, which actually doesn't use either of these proposed solves and I think is simpler and more general, without creating new enforced naming or argument conventions.

**Comment by whilei** [comment]
Resolved with #333
